collectors/plugins.d/pluginsd_parser.c

```diff
@@ -49,11 +49,16 @@ PARSER_RC pluginsd_chart_action(void *user, char *type, char *id, char *name, ch
 {
     RRDSET *st = NULL;
     RRDHOST *host = ((PARSER_USER_OBJECT *) user)->host;
+    uuid_t  *guid = &((PARSER_USER_OBJECT *)user)->guid;
 
-    st = rrdset_create(
-        host, type, id, name, family, context, title, units,
-        plugin, module, priority, update_every,
-        chart_type);
+    st = rrdset_create_custom(
+        host, type, id, name, family, context, title, units, plugin, module, priority, update_every, chart_type,
+        host->rrd_memory_mode, host->rrd_history_entries, 0,
+        uuid_is_null(*guid) ? NULL : guid);
+
+#ifdef ENABLE_DBENGINE
+    uuid_clear(((PARSER_USER_OBJECT *) user)->guid);
```

Comment:
Since plugins.d is performance critical it may be worth it to not burn CPU/RAM cycles unconditionally at this point. Maybe add something like `if (uuid_was_null)`?